src/storage/store/compression_flush_buffer.cpp

```diff
@@ -26,48 +25,45 @@ ColumnChunkMetadata CompressedFlushBuffer::operator()(const uint8_t* buffer,
     auto numPages = 0u;
-    const auto numValuesPerPage =
-        metadata.compMeta.numValues(BufferPoolConstants::PAGE_4KB_SIZE, dataType);
+    const auto numValuesPerPage = metadata.compMeta.numValues(PAGE_SIZE, dataType);
     KU_ASSERT(numValuesPerPage * metadata.numPages >= metadata.numValues);
     while (valuesRemaining > 0) {
         const auto compressedSize = alg->compressNextPage(bufferStart, valuesRemaining,
-            compressedBuffer.get(), BufferPoolConstants::PAGE_4KB_SIZE, metadata.compMeta);
+            compressedBuffer.get(), PAGE_SIZE, metadata.compMeta);
         // Avoid underflows (when data is compressed to nothing, numValuesPerPage may be
         // UINT64_MAX)
         if (numValuesPerPage > valuesRemaining) {
             valuesRemaining = 0;
         } else {
             valuesRemaining -= numValuesPerPage;
         }
-        if (compressedSize < BufferPoolConstants::PAGE_4KB_SIZE) {
-            memset(compressedBuffer.get() + compressedSize, 0,
-                BufferPoolConstants::PAGE_4KB_SIZE - compressedSize);
+        if (compressedSize < PAGE_SIZE) {
+            memset(compressedBuffer.get() + compressedSize, 0, PAGE_SIZE - compressedSize);
         }
         KU_ASSERT(numPages < metadata.numPages);
         KU_ASSERT(dataFH->getNumPages() > startPageIdx + numPages);
         if (dataFH->isInMemoryMode()) {
             const auto frame = dataFH->getFrame(startPageIdx + numPages);
-            memcpy(frame, compressedBuffer.get(), BufferPoolConstants::PAGE_4KB_SIZE);
+            memcpy(frame, compressedBuffer.get(), PAGE_SIZE);
         } else {
-            dataFH->getFileInfo()->writeFile(compressedBuffer.get(),
-                BufferPoolConstants::PAGE_4KB_SIZE,
-                (startPageIdx + numPages) * BufferPoolConstants::PAGE_4KB_SIZE);
+            dataFH->getFileInfo()->writeFile(compressedBuffer.get(), PAGE_SIZE,
+                (startPageIdx + numPages) * PAGE_SIZE);
         }
         numPages++;
     }
     // Make sure that the file is the right length
-    if (numPages < metadata.numPages) {
-        memset(compressedBuffer.get(), 0, BufferPoolConstants::PAGE_4KB_SIZE);
+    while (numPages < metadata.numPages) {
+        memset(compressedBuffer.get(), 0, PAGE_SIZE);
         if (dataFH->isInMemoryMode()) {
             const auto frame = dataFH->getFrame(startPageIdx + metadata.numPages - 1);
-            memcpy(frame, compressedBuffer.get(), BufferPoolConstants::PAGE_4KB_SIZE);
+            memcpy(frame, compressedBuffer.get(), PAGE_SIZE);
         } else {
-            dataFH->getFileInfo()->writeFile(compressedBuffer.get(),
-                BufferPoolConstants::PAGE_4KB_SIZE,
-                (startPageIdx + metadata.numPages - 1) * BufferPoolConstants::PAGE_4KB_SIZE);
+            dataFH->getFileInfo()->writeFile(compressedBuffer.get(), PAGE_SIZE,
+                (startPageIdx + metadata.numPages - 1) * PAGE_SIZE);
         }
+        numPages++;
```

Comment:
`numPages` isn't being used in the loop. It was just running once since writing to the last page will extend the file to that length. We shouldn't need to write to each individual page (and making this a loop just means that it writes the last page multiple times). I don't know if there is a noticeable performance benefit, but writing just the last page avoids the need for multiple system calls and the file system can handle the resizing internally.

I also have a suspicion that this code block is entirely unnecessary in in-memory-mode since the frames should default to zeroed and we've already reserved the pages in memory.

Reply:
Yeah make sense, only writing to the last page is needed. I got it wrong here.

> I also have a suspicion that this code block is entirely unnecessary in in-memory-mode since the frames should default to zeroed and we've already reserved the pages in memory.
Nice catch on this. I think so. Added an extra check to avoid the write under in-mem mode.